pennylane_lightning_gpu/src/algorithms/AdjointDiffGPU.hpp

```diff
@@ -18,188 +19,150 @@ template <class CFP_t> static constexpr auto getP11_CU() -> std::vector<CFP_t> {
 }
 
 template <class T = double, class SVType>
-void applyGeneratorRX_GPU(SVType &sv, const std::vector<size_t> &wires,
-                          const bool adj = false) {
+void applyGeneratorPauliRot_GPU(SVType &sv,
+                                const std::vector<std::string> &hyperparams,
+                                const std::vector<size_t> &wires,
+                                const bool adj = false) {
+    PL_ABORT_IF(hyperparams.size() != wires.size(),
+                "Incompatible number of Pauli words and wires");
+    for (size_t i = 0; i < wires.size(); i++) {
+        if (hyperparams[i] == "RX") {
+            sv.applyPauliX({wires[i]}, adj);
+        } else if (hyperparams[i] == "RY") {
+            sv.applyPauliY({wires[i]}, adj);
+        } else if (hyperparams[i] == "RZ") {
+            sv.applyPauliZ({wires[i]}, adj);
+        } else {
+            std::string message = "Unsupported Pauli word: " + hyperparams[i];
+            throw LightningException(message);
+        }
+    } // TODO: test & re-write this part
+}
+
+template <class T = double, class SVType>
+void applyGeneratorRX_GPU(
+    SVType &sv, [[maybe_unused]] const std::vector<std::string> &hyperparams,
```

Comment:
I think the addition of another arg is fine for now, but we can probably adapt this later (maybe using functors, or a visitor pattern) to only hold args where they can be used --- though this will require some changes in Lightning + LightningGPU.

Though, I think it would be better to use this as the last arg, as it can then be defaulted where applicable. What do you think?

Reply:
Nice suggestion! Yea I'll address it in a followup PR in Lightning and consequently here too.  For now, I updated the generators' signatures by moving `hyperparams` to the last arg.